parser/src/parser/input.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the Leo library.
+
+// The Leo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Leo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+
+use leo_errors::{ParserError, Result};
+
+impl ParserContext<'_> {
+    /// Returns a [`ParsedInputFile`] struct filled with the data acquired in the file.
+    pub fn parse_input(&mut self) -> Result<ParsedInputFile> {
+        let mut sections = Vec::new();
+        let mut is_public = true;
+
+        while self.has_next() {
+            let token = self.peek()?;
+            if matches!(token.token, Token::LeftSquare) {
+                // For visibility modifiers: [[public]] or [[private]]
+                if self.peek_next()?.token == Token::LeftSquare {
+                    is_public = self.parse_visibility()?;
+                }
+
+                let mut section = self.parse_section()?;
+                section.is_public = is_public;
+                sections.push(section);
+            } else {
+                return Err(ParserError::unexpected_token(token.token.clone(), &token.span).into());
+            }
+        }
+
+        Ok(ParsedInputFile { sections })
+    }
+
+    /// Parses visibility tables in the State or Input file.
+    /// Expects: [[<public>]] or [[<private>]], returns true
+    /// if visibility is set to public, returns else otherwise.
+    pub fn parse_visibility(&mut self) -> Result<bool> {
+        self.expect(Token::LeftSquare)?;
+        self.expect(Token::LeftSquare)?;
+        let visiblity = self.expect_ident()?;
+        self.expect(Token::RightSquare)?;
```

Comment:
Repeated in `parse_section`.

Reply:
Not quite. It's double square there `[[`.

Reply:
Do we need `[[` our old input file format only used `[ident]` rather than the toml style tables.

Reply:
Right. My understanding was also that we only need `[<identifier>]` sections here.

Reply:
Well, we have it currently in a default input files. I'm down to remove these!

Reply:
Ended up removing state file parsing.